### PR TITLE
chore: upgrade to Rust 1.67

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: cli/napi/async.rs [hunk 1]
```diff
@@ -39,6 +39,7 @@ fn napi_cancel_async_work(
 /// Frees a previously allocated work object.
 #[napi_sym::napi_sym]
 fn napi_delete_async_work(_env: &mut Env, work: napi_async_work) -> Result {
+  #[allow(clippy::from_raw_with_void_ptr)]
```
Comment (dsherret) [inline cli/napi/async.rs:42]: https://rust-lang.github.io/rust-clippy/master/index.html#from_raw_with_void_ptr

I feel like this shouldn't be allowed, but I'm not sure what to do here. cc @bartlomieju @littledivy

Comment (dsherret) [reply cli/napi/async.rs:42]: ` as *mut AsyncWork`?

**Review by ry — COMMENTED** [review]
How come this is not bumping a version anywhere?  I think we should be explicit about what version we're building against.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**